lib/Dec2Hex.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022-present, Trail of Bits, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed in accordance with the terms specified in
+ * the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <clang/AST/Expr.h>
+#include <clang/ASTMatchers/ASTMatchFinder.h>
+#include <clang/ASTMatchers/ASTMatchers.h>
+#include <clang/Basic/SourceManager.h>
+#include <clang/Rewrite/Core/Rewriter.h>
+#include <clang/Tooling/Tooling.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/raw_ostream.h>
+
+namespace rellic {
+namespace {
+using namespace clang;
+using namespace clang::ast_matchers;
+
+StatementMatcher intlit = integerLiteral().bind("intlit");
+
+class IntegerReplacer : public MatchFinder::MatchCallback {
+  Rewriter &rw;
+  std::function<bool(const llvm::APInt &)> shouldConvert;
+
+ public:
+  IntegerReplacer(Rewriter &rw,
+                  std::function<bool(const llvm::APInt &)> shouldConvert)
+      : rw(rw), shouldConvert(shouldConvert) {}
+  virtual void run(const MatchFinder::MatchResult &Result) {
+    if (auto lit = Result.Nodes.getNodeAs<IntegerLiteral>("intlit")) {
+      if (!shouldConvert(lit->getValue())) {
+        return;
+      }
+
+      llvm::SmallString<40> str;
+      lit->getValue().toString(
+          str, /*radix=*/16, /*isSigned=*/lit->getType()->isSignedIntegerType(),
+          /*formatAsCLiteral=*/true);
+      std::string res;
+      llvm::raw_string_ostream OS(res);
+      OS << str;
+      switch (lit->getType()->castAs<BuiltinType>()->getKind()) {
+        default:
+          llvm_unreachable("Unexpected type for integer literal!");
+        case BuiltinType::Char_S:
```

Comment:
Suffixes containing `i` like `Ui8` are and MSVC extension iirc, but I think it's okay to leave it here?

Reply:
This specific part is lifted straight from Clang, but I guess we could tune it to our liking if need be